toolkit/tools/pkg/imagecustomizerlib/bootcustomizer.go

```diff
@@ -160,37 +160,45 @@ func (b *BootCustomizer) UpdateKernelCommandLineArgs(defaultGrubFileVarName defa
 }
 
 // Makes changes to the /etc/default/grub file that are needed/useful for enabling verity.
-func (b *BootCustomizer) PrepareForVerity() error {
+func (b *BootCustomizer) PrepareForVerity(rootDeviceValue string) error {
 	if b.isGrubMkconfig {
 		// Force root command-line arg to be referenced by /dev path instead of by UUID.
-		defaultGrubFileContent, err := updateDefaultGrubFileVariable(b.defaultGrubFileContent, "GRUB_DISABLE_UUID",
+		defaultGrubFileContent, err := UpdateDefaultGrubFileVariable(b.defaultGrubFileContent, "GRUB_DISABLE_UUID",
 			"true")
 		if err != nil {
 			return err
 		}
 
 		// Disable recovery menu entry, to avoid having more than 1 linux command in the grub.cfg file.
 		// This will make it easier to modify the grub.cfg file to add the verity args.
-		defaultGrubFileContent, err = updateDefaultGrubFileVariable(defaultGrubFileContent, "GRUB_DISABLE_RECOVERY",
+		defaultGrubFileContent, err = UpdateDefaultGrubFileVariable(defaultGrubFileContent, "GRUB_DISABLE_RECOVERY",
 			"true")
 		if err != nil {
 			return err
 		}
 
+		// Check if rootDeviceValue is available as a field in BootCustomizer or another way
+		if rootDeviceValue != "" {
+			defaultGrubFileContent, err = UpdateDefaultGrubFileVariable(defaultGrubFileContent, "GRUB_DEVICE", rootDeviceValue)
```

Comment:
when I test EMU in trident, "/dev/mapper/root" is somehow not set in original /etc/default/grub, so I guess still need to read from grub.cfg

Reply:
If you are using an image generated by the existing MIC version, then `GRUB_DEVICE` won't be set in `/etc/default/grub` because your change hasn't gone in yet.

Please remove the `rootDeviceValue` parameter and always set `GRUB_DEVICE` to `"/dev/mapper/root"`.

If you want to continue calling `PrepareForVerity()` in EMU/Trident so that the code works with older images, then that is fine.